pkgs/build-support/zig/build-zig-package/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib, stdenv, zig }:
+
+{ strictDeps ? true
+, nativeBuildInputs ? [ ]
+, enableParallelBuilding ? true
+, meta ? { }
+, ...
+}@attrs:
+
+stdenv.mkDerivation (attrs // {
```

Comment:
**Don't apply**

```suggestion
stdenv.mkDerivation (finalAttrs: attrs // {
```

Ok, so I would have suggested this, and then reference `finalAttrs.zigBuildFlags` and such down below, but by going through the builder environment, you achieve the same effect.

If you do ever need to have a bit of Nix-level logic that involves a derivation attribute, you could make use of `finalAttrs` and get the final value from there, all at the Nix language level.

```nix
zig build -Drelease-safe -Dcpu=baseline ${frobnicate finalAttrs.zigBuildFlags}
```

Something like that.

Reply:
What's frobnicate

Reply:
Thanks for your time!

Reply:
> frobnicate

whatever you want it to be. I should've just said `foo`.

yw!

Reply:
> What's frobnicate

It's the _spam&eggs_ of Python documentation.